### PR TITLE
Add page info field to connection object

### PR DESCRIPTION
**Pull request by JunichiSugiura**
This PR adds `pageInfo` fields in all connection object exposed by Torii in order to enable server side pagination.

### CODE REVIEW

**Comment by JunichiSugiura** [comment]
### References

- https://stackoverflow.com/a/66300422
  - maybe i should get rid of `fetch_page_info` and and fetch N+2 queries on fetch_multiple_rows

**Comment by JunichiSugiura** [comment]
@broody How can I return Null for nested object?

Current error

Screenshot 2023-11-18 at 16 29 24

Because I'm returning `Value::Null` [here](https://github.com/dojoengine/dojo/blob/d4078a5c19b8fae942856ef181f117609aaa5658/crates/torii/graphql/src/object/connection/mod.rs#L146) I think.

**Review by broody — COMMENTED** [review]
> @broody How can I return Null for nested object?

Ah this is an issue, in the schema, nested objects are by default nullable but then we reject null values. I'll fix this, but for `page_info` I think we do want a non-null value object.

File: crates/torii/graphql/src/object/connection/mod.rs [hunk 1]
```diff
@@ -37,6 +41,10 @@ impl ConnectionObject {
                 TypeData::Simple(TypeRef::named_list(format!("{}Edge", type_name))),
             ),
             (Name::new("total_count"), TypeData::Simple(TypeRef::named_nn(TypeRef::INT))),
+            (
+                Name::new("page_info"),
+                TypeData::Nested((TypeRef::named(PAGE_INFO_TYPE_NAME), IndexMap::new())),
```
[the remark this answers is not in the document]
Comment (JunichiSugiura) [reply crates/torii/graphql/src/object/connection/mod.rs:46]: Thanks for the info. I wonder this field is useful for query with `limit/offset` params but i can still provide `has_*_page` fields with cursor of null.

**Comment by JunichiSugiura** [comment]
Should be good now !

**Review by broody — APPROVED** [review]

**Pull request merged**